packages/commonwealth/server/passport/magic.ts

```diff
@@ -20,171 +20,215 @@ export function useMagicAuth(models: DB) {
+                issuer: userMetadata.issuer,
+                issued_at: user.claim.iat,
+                address_id: newAddress.id, // always ethereum address
+                created_at: new Date(),
+                updated_at: new Date(),
+              },
+              { transaction: t }
+            );
+
+            return newUser;
+          });
+
+          // re-fetch user to include address object
+          const newUser = await models.User.findOne({
+            where: {
+              id: result.id,
+            },
+            include: [models.Address],
+          });
+          return cb(null, newUser);
+        } else if (existingUser.Addresses?.length > 0) {
+          // each user should only ever have one token issued by Magic
+          const ssoToken = await models.SsoToken.findOne({
+            where: {
+              issuer: user.issuer,
+            },
+            include: [
+              {
+                model: models.Address,
+                where: { address: user.publicAddress },
+                required: true,
+              },
+            ],
+          });
+          // login user if they registered via magic
+          if (user.claim.iat <= ssoToken.issued_at) {
+            console.log('Replay attack detected.');
+            return cb(null, null, {
+              message: `Replay attack detected for user ${user.publicAddress}}.`,
+            });
+          }
+          ssoToken.issued_at = user.claim.iat;
+          ssoToken.updated_at = new Date();
+          await ssoToken.save();
+          console.log(`Found existing user: ${JSON.stringify(existingUser)}`);
+          return cb(null, existingUser);
+        } else {
+          // migrate to magic if email already exists
+          await sequelize.transaction(async (t) => {
+            const ethAddress = userMetadata.publicAddress;
+            const newAddress = await models.Address.create(
+              {
+                address: ethAddress,
+                chain: 'ethereum',
+                verification_token: 'MAGIC',
+                verification_token_expires: null,
+                verified: new Date(), // trust addresses from magic
+                last_active: new Date(),
+                user_id: existingUser.id,
+                profile_id: (existingUser.Profiles[0] as ProfileAttributes).id,
+                wallet_id: WalletId.Magic,
+              },
+              { transaction: t }
+            );
+
+            await models.SsoToken.create(
+              {
+                issuer: userMetadata.issuer,
+                issued_at: user.claim.iat,
+                address_id: newAddress.id, // always ethereum address
+                created_at: new Date(),
+                updated_at: new Date(),
+              },
+              { transaction: t }
+            );
+          });
+          console.log(
+            `Migrated old email user to magic: ${existingUser.email}`
+          );
+          return cb(null, existingUser);
```

Comment:
What happened with this file? Shouldn't have been changed at all.

Reply:
reverting

Reply:
actually the reason why this is changed because I moved validateChain into the middleware folder!

Reply:
Still not clear to me -- can you explain exactly what changes happened? Remember that passport logic is not the same as normal route logic -- it's also middleware.

Reply:
I see now, I think it is due to the weird linter I have, let me roll this back to the original but change the import